### ISSUE TITLE
doom upgrade broken with core v3.0.0-dev and modules v22.06.0-dev

### ISSUE THREAD

**Issue by liar666** [doomemacs/doomemacs#6532]
### What did you expect to happen?

`doom upgrade` to proceed & succeed

### What actually happened?

```
$ doom upgrade
> Preparing to upgrade Doom Emacs and its packages...
✓ Doom is already up-to-date!
Doom is up-to-date!
> Compiling your literate config...
x There was an unexpected runtime error
  Message: Symbol's function definition is void
  Details: (always)
  Backtrace:
    (always (doom-print (doom-print--format "Restarting...") :format nil))
    (and (not (getenv "__NOTANGLE")) (+literate-tangle +literate-config-file (concat doom-module-config-file ".el") doom-private-...
    (+literate-tangle--sync)
    (if noninteractive (+literate-tangle--sync) (+literate-tangle--async))
    (+literate-tangle-h)
    (run-hooks doom-before-sync-hook)
    (let ((noenvvar\? (cdr (assq 'noenvvar\? alist))) (noelc\? (cdr (assq 'noelc\? alist))) (update\? (cdr (assq 'update\? alist)...
    ((closure (t) (cli alist) (let ((noenvvar\? (cdr (assq 'noenvvar\? alist))) (noelc\? (cdr (assq 'noelc\? alist))) (update\? (...
    (funcall (closure (t) (cli alist) (let ((noenvvar\? (cdr (assq 'noenvvar\? alist))) (noelc\? (cdr (assq 'noelc\? alist))) (up...
    (doom-cli-execute #s(doom-cli ("doom" "sync") nil (("SUMMARY" . "Synchronize your config with Doom Emacs.") ("MAIN" . "This i...
    (let ((bindings x5) (cli x4)) (doom-cli-execute cli bindings))
    (let* ((x4 (car x3)) (x5 (cdr x3))) (let ((bindings x5) (cli x4)) (doom-cli-execute cli bindings)))
  ! Wrote extended backtrace to ~/.emacs.doom.d/.local/logs/cli.doom.220704101212.725787.error
✓ Finished in 1.52441s
```

The quoted stacktrace is attached
[error.txt](https://github.com/doomemacs/doomemacs/files/9037683/error.txt)

### Describe your attempts to resolve the issue

I don't understand a word from the error message/log, so I haven't tried so much.

Only thing I understand is that error seems related to my `config.org/.el`. I removed them completely. Same error.

I've tried  `doom upgrade` several times. I've tried `git pull` directly in the `~/.emacs.d` dir but it is already as fresh as possible.

I've tried `doom clean` then `doom upgrade` with no success

### Steps to reproduce

Not sure if it is specific to my config or generic. just using `doom upgrade` in my preconfigured directory gives the error.
Maybe one should try to create a fresh doom install with a previous version, then create a `config.org` then try to upgrade. But I'm not sure how to achieve that without breaking my current config

FYI, I've seen another issue with doom compile, so I tried it too and it also result in an error:
```
$ doom compile
WARNING: Changes made to your config after compiling it won't take effect until
this command is rerun or you run 'doom clean'! It will also make error backtraces
much more difficult to decipher.

If you intend to use it anyway, remember this or it will come back to bite you!

Continue anyway? (y or n) y
x There was an unexpected runtime error
  Message: Symbol's value as variable is void
  Details: (doom-interactive-p)
  Backtrace:
    (defvar projectile-enable-caching doom-interactive-p)
    (load "/home/user/.emacs.doom.d/.local/autoloads.27.1" nil nomessage)
    (condition-case e (load (string-remove-suffix ".el" doom-autoloads-file) nil 'nomessage) ((debug file-missing) (if (locate-fi...
    (eval-buffer #<buffer  *load*-913292> nil "/home/user/.emacs.doom.d/core/core-start.el" nil t)
    (load-with-code-conversion "/home/user/.emacs.doom.d/core/core-start.el" "/home/user/.emacs.doom.d/core/core-start.el" ...
    (require core-start)
    (let ((load-prefer-newer t) (noninteractive t)) (require 'core-start) (if init-file-debug (progn (doom-initialize-packages)) ...
    (let ((after-load-functions (if (null targets) after-load-functions (cons (let ((target-dirs (seq-filter #'file-directory-p t...
    (let* ((default-directory doom-emacs-dir) (targets (nreverse (delete-dups targets))) (auto-mode-alist auto-mode-alist) kill-e...
    (doom-cli-compile ("/home/user/.emacs.doom.d/core/") nil nil)
    (let ((recompile-p (cdr (assq 'recompile-p alist))) (core-p (cdr (assq 'core-p alist))) (private-p (cdr (assq 'private-p alis...
    ((closure (t) (cli alist) (let ((recompile-p (cdr (assq 'recompile-p alist))) (core-p (cdr (assq 'core-p alist))) (private-p ...
  ! Wrote extended backtrace to ~/.emacs.doom.d/.local/logs/cli.doom.220704101000.724643.error

```

### System Information

https://pastebin.com/0ej3LcL3

**Comment by liar666** [comment]
Arg.... Now even just trying to open Doom Results in 
 ```
Warning (initialization): An error occurred while loading ‘/home/user/.emacs.d/core/core-start.el’:

Symbol's value as variable is void: doom-interactive-p
```

**Comment by hlissner** [comment]
The original `void-function: always` error has been fixed in 38a3a50. You may need to upgrade manually if the recent string of `doom upgrade`'s left your install in a bad spot:

```sh
cd ~/.emacs.d
git pull --rebase
doom sync -u
```
